domain/src/test/java/org/folio/spring/domain/controller/JsonSchemasControllerTest.java

```diff
@@ -0,0 +1,243 @@
+import static org.folio.spring.test.mock.MockMvcConstant.OKAPI_HEAD;
+import static org.folio.spring.test.mock.MockMvcConstant.OKAPI_HEAD_NO_URL;
+import static org.folio.spring.test.mock.MockMvcConstant.PATH_PARAM;
+import static org.folio.spring.test.mock.MockMvcConstant.PLAIN_BODY;
+import static org.folio.spring.test.mock.MockMvcConstant.STAR;
+import static org.folio.spring.test.mock.MockMvcConstant.STRING_LIST;
+import static org.folio.spring.test.mock.MockMvcConstant.STRING_LIST_AS_JSON;
+import static org.folio.spring.test.mock.MockMvcConstant.TEXT_PLAIN;
+import static org.folio.spring.test.mock.MockMvcReflection.DELETE;
+import static org.folio.spring.test.mock.MockMvcReflection.PATCH;
+import static org.folio.spring.test.mock.MockMvcReflection.POST;
+import static org.folio.spring.test.mock.MockMvcReflection.PUT;
+import static org.folio.spring.test.mock.MockMvcRequest.appendBody;
+import static org.folio.spring.test.mock.MockMvcRequest.appendHeaders;
+import static org.folio.spring.test.mock.MockMvcRequest.appendParameters;
+import static org.folio.spring.test.mock.MockMvcRequest.buildArguments1;
+import static org.folio.spring.test.mock.MockMvcRequest.buildArguments2;
+import static org.folio.spring.test.mock.MockMvcRequest.invokeRequestBuilder;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.lenient;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import org.folio.spring.domain.service.JsonSchemasService;
+import org.folio.spring.test.mock.MockMvcConstant;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.util.MultiValueMap;
+
+@SpringBootTest
```

Comment:
Can we use [WebMvcTest](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/autoconfigure/web/servlet/WebMvcTest.html) here?

Reply:
What for?
And no, switching to the `@WebMvcTest` annotation and it fails spectacularly.

Reply:
>Annotation that can be used for a Spring MVC test that focuses only on Spring MVC components.

Reply:
Is that annotation no longer preferred to ensure isolation of the controller test? Or whatever reason Spring Framework has created the annotation for.